### DISCUSSION TITLE
How exactly does DurableFunctionsMonitor.DotNetBackend package 'inject' itself into your Function App project?

### DISCUSSION THREAD

**Discussion by drdamour** [scale-tone/DurableFunctionsMonitor#102] [Q&A]
it's been on my list to understand how to package some functions for inclusion in other function apps, and wanted to pick your brain if i could.

re: 
https://github.com/scale-tone/DurableFunctionsMonitor/blob/93b60aa7fff514ebd50f4821a0aa38d73c4f9f9a/durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.targets#L6

Does this make functions scan ALL dependencies for functions, or just this assembly?

re: 
https://github.com/scale-tone/DurableFunctionsMonitor/blob/93b60aa7fff514ebd50f4821a0aa38d73c4f9f9a/durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.targets#L96
why is this second task needed? does the publish task recreate the functions.json or does it sometimes skip the postbuildtask on a publish?

re:
https://github.com/scale-tone/DurableFunctionsMonitor/blob/93b60aa7fff514ebd50f4821a0aa38d73c4f9f9a/durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.targets#L13

i'm very confused how the assets works... they obv get packages in the nuget 
https://github.com/scale-tone/DurableFunctionsMonitor/blob/93b60aa7fff514ebd50f4821a0aa38d73c4f9f9a/durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.csproj#L33-L36

and i think this means that they get put in the output of my build https://github.com/scale-tone/DurableFunctionsMonitor/blob/93b60aa7fff514ebd50f4821a0aa38d73c4f9f9a/durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.csproj#L33-L36

so does
https://github.com/scale-tone/DurableFunctionsMonitor/blob/93b60aa7fff514ebd50f4821a0aa38d73c4f9f9a/durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.targets#L13

pull them from my output dir?

**Comment by scale-tone** [comment]
> FunctionsInDependencies
> Does this make functions scan ALL dependencies for functions, or just this assembly?

That setting is project-wide, it's not tied to any particular assembly, so I assume it scans all dependencies.

> why is this second task needed? does the publish task recreate the functions.json or does it sometimes skip the postbuildtask on a publish?

Publish outputs its artifacts into a separate folder (`$(PublishDir)`), and function.json files there appear to be fresh new. AFAIR, those files are produced by a separate post-build/post-publish task, not the build task itself. So DfMon needs to run FixDfmPrefixesInFunctionJsonAfterPublish on those files as well.

> so does durablefunctionsmonitor.dotnetbackend.targets pull them from my output dir?

When you install a NuGet-package, only its binaries (more precisely, the binaries inside its `lib` folder) get automatically copied to your project. So DfMon's statics need to be copied _from package to your project_ separately, and that's what line 13 does.

**Comment by scale-tone** [comment]
Hope that helps :)
